src/components/settings/SettingsSSOTab/steps/NewSSOConfigConfigureStep.tsx

```diff
@@ -4,50 +4,84 @@ import {
 } from '@edx/paragon';
 
 import ValidatedFormControl from '../../../forms/ValidatedFormControl';
+import { FormContext, FormFieldValidation, useFormContext } from '../../../forms/FormContext';
+import { urlValidation } from '../../../../utils';
+
+const isSAPConfig = (fields) => fields.identityProvider === 'sap_success_factors';
+
+export const validations: FormFieldValidation[] = [
+  {
+    formFieldId: 'sapsfOauthRootUrl',
+    validator: (fields) => isSAPConfig(fields) && (!fields.sapsfOauthRootUrl || !urlValidation(fields.sapsfOauthRootUrl)) && 'Please enter an OAuth Root URL.',
+  },
+  {
+    formFieldId: 'odataApiRootUrl',
+    validator: (fields) => isSAPConfig(fields) && (!fields.odataApiRootUrl || !urlValidation(fields.odataApiRootUrl)) && 'Please enter an API Root URL.',
+  },
+  {
+    formFieldId: 'sapsfPrivateKey',
+    validator: (fields) => isSAPConfig(fields) && !fields.sapsfPrivateKey && 'Please enter a Private Key.',
+  },
+  {
+    formFieldId: 'odataCompanyId',
+    validator: (fields) => isSAPConfig(fields) && !fields.odataCompanyId && 'Please enter a Company ID.',
+  },
+  {
+    formFieldId: 'oauthUserId',
+    validator: (fields) => isSAPConfig(fields) && !fields.oauthUserId && 'Please enter an OAuth User ID.',
+  },
+];
 
 const SSOConfigConfigureStep = () => {
+  const {
+    formFields,
+  }: FormContext = useFormContext();
+  const usingSAP = formFields?.identityProvider === 'sap_success_factors';
+
   const renderBaseFields = () => (
     <>
-      <h3>Enter user attributes</h3>
+      <span className="d-flex pb-4">
+        <h3>Enter user attributes</h3>
+      </span>
       <p>
         Please enter the SAML user attributes from your Identity Provider.
         All attributes are space and case sensitive.
       </p>
       <Form.Group className="mb-4">
         <ValidatedFormControl
-          formId="DummyOAuthUserID"
+          formId="userIdAttribute"
           type="text"
           floatingLabel="User ID"
           fieldInstructions="URN of the SAML attribute that edX can use as a unique, persistent user ID."
         />
       </Form.Group>
       <Form.Group className="mb-4">
         <ValidatedFormControl
-          formId="DummyOAuthFullName"
+          formId="fullNameAttribute"
           type="text"
           floatingLabel="Full Name"
           fieldInstructions="URN of SAML attribute containing the user's full name."
         />
       </Form.Group>
       <Form.Group className="mb-4">
         <ValidatedFormControl
-          formId="DummyOAuthFirstName"
+          formId="usernameAttribute"
```

Comment:
Still confused, shouldn't this be `formId="firstNameAttribute"`?

Reply:
username is a valid field we're looking for, but we do need a [first name attribute field](https://gist.github.com/praveenram2u/a31b3e33b7ef4f5e550a8f196bb61aed#file-sso-self_service_orchestration_api-yaml-L108) as well @marlonkeating !

Reply:
Added!